### PR TITLE
show groupings and swallowing in hyprctl

### PR DESCRIPTION
**Pull request by eriedaberrie**
#### Describe your PR, what does it fix/add?
Shows grouped and swallowed windows in the hyprctl client data.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
No.

#### Is it ready for merging, or does it need work?
Looks good for merging to me, but this is my first time touching a C++ project with more than one source file so let me know if there's anything I did bad I should change.

### CODE REVIEW

File: src/debug/HyprCtl.cpp [hunk 1]
```diff
@@ -101,11 +128,13 @@ R"#({
                     ((int)w->m_bIsX11 == 1 ? "true" : "false"),
                     (w->m_bPinned ? "true" : "false"),
                     (w->m_bIsFullscreen ? "true" : "false"),
-                    (w->m_bIsFullscreen ? (g_pCompositor->getWorkspaceByID(w->m_iWorkspaceID) ? g_pCompositor->getWorkspaceByID(w->m_iWorkspaceID)->m_efFullscreenMode : 0) : 0)
+                    (w->m_bIsFullscreen ? (g_pCompositor->getWorkspaceByID(w->m_iWorkspaceID) ? g_pCompositor->getWorkspaceByID(w->m_iWorkspaceID)->m_efFullscreenMode : 0) : 0),
+                    (g_pLayoutManager->getCurrentLayout()->getLayoutName() == "dwindle" ? getGroupedData(w, format).c_str() : ""),
+                    (w->m_pSwallowed ? w->m_pSwallowed->getPID() : 0)
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:133]: using PID for swallowed kinda sucks. Why not use the handle? `w.get()` and then %x in `"swallowing"` above

File: src/debug/HyprCtl.cpp [hunk 2]
```diff
@@ -101,11 +128,13 @@ R"#({
                     ((int)w->m_bIsX11 == 1 ? "true" : "false"),
                     (w->m_bPinned ? "true" : "false"),
                     (w->m_bIsFullscreen ? "true" : "false"),
-                    (w->m_bIsFullscreen ? (g_pCompositor->getWorkspaceByID(w->m_iWorkspaceID) ? g_pCompositor->getWorkspaceByID(w->m_iWorkspaceID)->m_efFullscreenMode : 0) : 0)
+                    (w->m_bIsFullscreen ? (g_pCompositor->getWorkspaceByID(w->m_iWorkspaceID) ? g_pCompositor->getWorkspaceByID(w->m_iWorkspaceID)->m_efFullscreenMode : 0) : 0),
+                    (g_pLayoutManager->getCurrentLayout()->getLayoutName() == "dwindle" ? getGroupedData(w, format).c_str() : ""),
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:132]: you can move the dwindle check to the function. Here it's just clutter

File: src/debug/HyprCtl.cpp [hunk 3]
```diff
@@ -68,7 +68,35 @@ R"#({
...
+        else
+            result << comma;
+        result << gw->getPID();
+    }
+
+    return result.str();
+}
+
 static std::string getWindowData(CWindow* w, HyprCtl::eHyprCtlOutputFormat format) {
+    SLayoutMessageHeader header;
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:97]: you moved these to the func, unused

**Comment by eriedaberrie** [comment]
Do you think grouped should also be returning handles instead of pids?

**Comment by vaxerski** [comment]
> Do you think grouped should also be returning handles instead of pids?

absolutely

**Comment by vaxerski** [comment]
thanks!

**Pull request merged**